src/transport/subscription-worker/components/heartbeat-state.ts

```diff
@@ -186,13 +198,27 @@ export class HeartbeatState extends EventTarget {
     this.requests[client.identifier] = request;
     this.clientsState[client.identifier] = { channels: request.channels, channelGroups: request.channelGroups };
     if (request.state) this.clientsState[client.identifier].state = { ...request.state };
+    const presenceState = this.clientsPresenceState[client.identifier];
+    const cachedPresenceStateKeys = presenceState ? Object.keys(presenceState.state) : [];
+
+    if (presenceState && cachedPresenceStateKeys.length) {
+      cachedPresenceStateKeys.forEach((key) => {
+        if (!request.channels.includes(key) && !request.channelGroups.includes(key)) delete presenceState.state[key];
+      });
+
+      if (Object.keys(presenceState.state).length === 0) delete this.clientsPresenceState[client.identifier];
+    }
 
     // Update access token information (use the one which will provide permissions for longer period).
     const sortedTokens = Object.values(this.requests)
       .filter((request) => !!request.accessToken)
       .map((request) => request.accessToken!)
       .sort(AccessToken.compare);
-    if (sortedTokens && sortedTokens.length > 0) this._accessToken = sortedTokens.pop();
+    if (sortedTokens && sortedTokens.length > 0) {
```

Comment:
Does AccessToken compare based on current time to it's ttl? And that's the only parameter?